src/main.js

```diff
@@ -396,32 +421,47 @@ function drawSelectionArea(e){
             .attr("height", 0)
         //console.log("selection is created");
     }
-    selection_area
-        .attr("width", e.pageX - selection_area.attr("x"))
-        .attr("height", e.pageY - selection_area.attr("y"));
+    var rectX = Number(selection_area.attr("x"));
+    var rectY = Number(selection_area.attr("y"));
+    var width = Number(selection_area.attr("width"));
+    var height = Number(selection_area.attr("height"));
+    var midX = rectX + width / 2;
+    var midY = rectY + height / 2;
+
+    if( e.pageX > midX ){
+      selection_area.attr("width", e.pageX - rectX);
+    } else{
+      selection_area.attr("width", rectX - e.pageX + width);
+      selection_area.attr("x", e.pageX);
+    }
+    if( e.pageY > midY ){
+      selection_area.attr("height", e.pageY - rectY);
+    } else{
+      selection_area.attr("height", rectY - e.pageY + height);
+      selection_area.attr("y", e.pageY);
+    }
```

Comment:
Can this be separated into new functions called getSelectionWidth() and getSelectionHeight()? Both of them will return value of width/height and whether it should set x/y.